pkg/storage/leveldbstore/transaction.go

```diff
@@ -79,17 +190,15 @@ func (s *TxStore) Put(item storage.Item) error {
 func (s *TxStore) Delete(item storage.Item) error {
 	err := s.TxStoreBase.Delete(item)
 	if err == nil {
-		s.revOps.Append(&storage.TxRevertOp{
+		val, merr := item.Marshal()
```

Comment:
We should do a Get before we call Marshal on the item. This follows the comment I had in the internal/reserve.go file. The item is not populated in most cases for Delete. Also, the user would have to do a Get anyway if he has to send it filled.

Reply:
Done.